frontend/src/components/Order/components/OrderInfo/OrderInfo.jsx

```diff
@@ -26,9 +32,66 @@ const OrderInfo = ({ orderData }) => {
     return quantity;
   }, [cartItems]);
 
-  // const sendOrder = () => {
-  //   console.log(cartItems, orderData);
-  // };
+  const sendOrder = async () => {
+    const orderItems = cartItems.map((item) => {
+      if (item.orderItemsCompositions) {
+        const bouquetItems = item.orderItemsCompositions.map(
+          (compositionItem) => {
+            return {
+              productId: compositionItem.id,
+              qty: compositionItem.cardQuantity,
+            };
+          },
+        );
+        return {
+          productTitle: item.title,
+          price: item.price,
+          qty: item.cardQuantity,
+          orderItemsCompositions: bouquetItems,
+        };
+      } else
+        return {
+          productId: item.id,
+          productTitle: item.title,
+          price: item.price,
+          qty: item.cardQuantity,
+        };
+    });
+
+    if (orderData.contactData.recipient === 'I') {
+      receiverName = orderData.contactData.clientFirstName;
+      receiverPhone = orderData.contactData.clientPhone;
+    } else {
+      receiverName =
+        orderData.contactData.recipientLastName +
+        ' ' +
+        orderData.contactData.recipientFirstName +
+        ' ' +
+        orderData.contactData.recipientMiddleName;
+      receiverPhone = orderData.contactData.recipientPhone;
+    }
+
+    const result = await addOrderToDB({
+      postcardText: orderData.postcardData.postcardText,
```

Comment:
Maybe it's better to use destructuring here. The code seems too complex

Reply:
Done